include/wlr/types/wlr_surface.h

```diff
@@ -29,6 +29,12 @@ enum wlr_surface_state_field {
 	WLR_SURFACE_STATE_VIEWPORT = 1 << 8,
 };
 
+struct wlr_surface_state_addon {
+	void *state;
```

Comment:
Why not `wlr_xdg_surface_state` ?

Reply:
The surface can have multiple extensions to it, not just `wlr_xdg_surface`.